packages/enhanced-img/src/preprocessor.js

```diff
@@ -49,20 +51,26 @@ export function image(opts) {
 			const consts = new Map();
 
 			/**
-			 * @param {import('svelte/types/compiler/interfaces').TemplateNode} node
-			 * @param {{ type: string, start: number, end: number, raw: string }} src_attribute
+			 * @param {import('svelte/compiler').AST.RegularElement} node
+			 * @param {AST.Text | AST.ExpressionTag} src_attribute
 			 * @returns {Promise<void>}
 			 */
 			async function update_element(node, src_attribute) {
-				// TODO: this will become ExpressionTag in Svelte 5
-				if (src_attribute.type === 'MustacheTag') {
-					const src_var_name = content
-						.substring(src_attribute.start + 1, src_attribute.end - 1)
-						.trim();
-					s.update(node.start, node.end, dynamic_img_to_picture(content, node, src_var_name));
-					return;
-				} else if (src_attribute.type === 'AttributeShorthand') {
-					const src_var_name = content.substring(src_attribute.start, src_attribute.end).trim();
+				if (src_attribute.type === 'ExpressionTag') {
+					const start =
+						'end' in src_attribute.expression
+							? src_attribute.expression.end
+							: src_attribute.expression.range?.[0];
+					const end =
+						'start' in src_attribute.expression
+							? src_attribute.expression.start
+							: src_attribute.expression.range?.[1];
+
+					if (typeof start !== 'number' || typeof end !== 'number') {
+						throw new Error('ExpressionTag has no range');
+					}
```

Comment:
This is technically more correct although I'm unsure we would ever realistically hit this issue. It was required for typescript to shut up tho.

The AST relating to locations did change very slightly with v5s modern AST, using `src_attribute.start` and `src_attribute.end` was incorrect.